modules/gdprEnforcement.js

```diff
@@ -157,28 +187,78 @@ export function userIdHook(fn, submodules, consentData) {
   }
 }
 
-const hasPurpose1 = (rule) => { return rule.purpose === purpose1 }
+/**
+ * Checks if a bidder is allowed. If it's not allowed, the bidder adapter won't send request to their endpoint.
+ * Enforces "purpose 2 (basic ads)" of TCF v2.0 spec
+ * @param {Function} fn - Function reference to the original function.
+ * @param {Array<adUnits>} adUnits
+ */
+export function makeBidRequestsHook(fn, adUnits, ...args) {
+  const consentData = gdprDataHandler.getConsentData();
+  if (consentData && consentData.gdprApplies) {
+    if (consentData.apiVersion === 2) {
+      const disabledBidders = [];
+      adUnits.forEach(adUnit => {
+        adUnit.bids = adUnit.bids.filter(bid => {
+          const currBidder = bid.bidder;
+          const gvlId = getGvlid(currBidder);
+          if (includes(disabledBidders, currBidder)) return false;
+          const isAllowed = gvlId && validateRules(purpose2Rule, consentData, currBidder, gvlId);
+          if (!isAllowed) {
+            utils.logWarn(`User blocked bidder: ${currBidder}. No bid request will be sent to their endpoint.`);
+            events.emit(EVENTS.BIDDER_BLOCKED, currBidder);
+            disabledBidders.push(currBidder);
+          }
+          return isAllowed;
+        });
+      });
+      fn.call(this, adUnits, ...args);
+    } else {
+      utils.logInfo('Enforcing TCF2 only');
```

Comment:
Suggest not logging anything here. Log messages take up space, and a comment would suffice, IMO: "// we don't enforce TCF1.1 strings".